code/go/0chain.net/smartcontract/dbs/event/process.go

```diff
@@ -49,27 +49,34 @@ const (
 )
 
 func (edb *EventDb) AddEvents(ctx context.Context, events []Event) {
-	newEvents := edb.removeDuplicate(ctx, events)
+	edb.eventsChannel <- events
+}
+
+func (edb *EventDb) addEventsWorker(ctx context.Context) {
+	for {
+		events := <-edb.eventsChannel
+		newEvents := edb.removeDuplicate(ctx, events)
```

Comment:
Were you able to do a performance benchmark of this query? As of now we are traversing all the events to check for duplicates, also it since in earlier query too we had this removeDuplicate check but then again in the logs we saw a lot of errors for unique-constraint violation.

Reply:
This optimization is not a part of this pr The query pr is https://github.com/0chain/0chain/pull/836